packages/grafana-data/src/transformations/transformers/joinDataFrames.ts

```diff
@@ -272,16 +281,22 @@ function nullExpand(yVals: Array<number | null>, nullIdxs: number[], alignedLen:
 }
 
 // nullModes is a tables-matched array indicating how to treat nulls in each series
-export function join(tables: AlignedData[], nullModes?: number[][]) {
-  const xVals = new Set<number>();
+export function join(tables: AlignedData[], nullModes?: number[][], mode: JoinMode = JoinMode.outer) {
+  let xVals: Set<number>;
 
-  for (let ti = 0; ti < tables.length; ti++) {
-    let t = tables[ti];
-    let xs = t[0];
-    let len = xs.length;
+  if (mode === JoinMode.inner) {
+    xVals = new Set(intersect(tables.map((t) => Array.from(t[0]))));
```

Comment:
i dont think we need to make copies of `t[0]` here since intersect doesnt seem to mutate the input?

https://github.com/lovasoa/fast_array_intersect/blob/4abccc64719b639849e6e5d87c2b6a1f5b9f38b3/index.ts#L14

```suggestion
    xVals = new Set(intersect(tables.map((t) => t[0])));
```

if we do need to make copies, `t[0].slice()` is probably faster:

```suggestion
    xVals = new Set(intersect(tables.map((t) => t[0].slice())));
```

Reply:
This was done to make types compatible in order to use `intersect`.
Without `Array.from` - seeing `Type 'number[] | TypedArray' is not assignable to type 'number[]'`.
And `slice` will lead to additional type issues.
So could keep as `Array.from` or update to something like `t[0] as number[]`

Reply:
i think type coercion or something like //@ts-ignore should be fine here. Grafana doesnt use typed arrays here anyways, and i think even in that case what intersect does internally would work for typed arrays.